ethstorage/p2p/discovery.go

```diff
@@ -149,6 +166,30 @@ func (v *Secp256k1) DecodeRLP(s *rlp.Stream) error {
 	return nil
 }
 
+func addNATMappings(tcpPort, udpPort int) (net.IP, int, int, error) {
```

Comment:
We may want to investigate whether libp2p can do this work if we enable NAT option in the libp2p constructor

Reply:
NAT manager is not public, we cannot directly use it.